src/jdk.jlink/share/classes/jdk/tools/jlink/internal/plugins/DefaultCompressPlugin.java

```diff
@@ -102,6 +103,13 @@ public void configure(Map<String, String> config) {
                     zip = new ZipPlugin(resFilter);
                     break;
                 default:
+                    if(level.length() == 5 && level.startsWith("zip-")) {
+                        try {
+                            int zipLevel = Integer.parseInt(level.substring(4));
+                            zip = new ZipPlugin(resFilter, zipLevel);
+                            break;
+                        } catch (NumberFormatException ignored) {}
```

Comment:
Hello Ian, previously before this change (and even now for non `zip-` values) we throw an `IllegalArgumentException` if the value for compression level is incorrect. Should we do the same for wrong values of `zip-` and throw `IllegalArgumentException` when we catch a `NumberFormatException`?

Reply:
Yes, though in this case when we catch a NumberFormatException, the result falls through to the IllegalArgumentException and prints the offending level ("zip-*") value.